org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/javadoc/JavaDocHTMLPathHandler.java

```diff
@@ -0,0 +1,355 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.javadoc;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileTime;
+import java.util.jar.JarFile;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.zip.ZipEntry;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.eclipse.core.internal.runtime.InternalPlatform;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.core.dom.TextElement;
+import org.eclipse.jdt.ls.core.internal.IConstants;
+
+/**
+ * @author Nikolas Komonen - nkomonen@redhat.com
+ *
+ */
+public class JavaDocHTMLPathHandler {
+
+	// path to extract images to
+	public static final String PLUGIN_TEMP_FOLDER = (SystemUtils.IS_OS_WINDOWS ? "/" : "") + InternalPlatform.getDefault().getStateLocation(Platform.getBundle(IConstants.PLUGIN_ID)).toPortableString() + "/";
+
+	// absolute path to folder holding all extracted images
+	public static final String EXTRACTED_JAR_IMAGES_FOLDER = PLUGIN_TEMP_FOLDER + "extracted-jar-images/";
```

Comment:
public static final IPath EXTRACTED_JAR_IMAGES_FOLDER = JavaLanguageServerPlugin.getInstance().getStateLocation().append("extracted-jar-images");

Reply:
This requires me to redo all the steps from PLUGIN_TEMP_FOLDER, and the only use case for it is as a string.